client/me/purchases/manage-purchase/purchase-meta.jsx

```diff
@@ -223,11 +225,51 @@ function PurchaseMetaOwner( { owner } ) {
 
 function PurchaseMetaPrice( { purchase } ) {
 	const translate = useTranslate();
-	const { productSlug, productDisplayPrice } = purchase;
-	const plan = getPlan( productSlug ) || getProductFromSlug( productSlug );
+	const { productDisplayPrice } = purchase;
+	const productsList = useSelector( getProductsList );
 	let period = translate( 'year' );
 
 	if ( isOneTimePurchase( purchase ) || isDomainTransfer( purchase ) ) {
+		if ( isDIFMProduct( purchase ) ) {
+			const productDetails = productsList[ WPCOM_DIFM_LITE ];
+			const [ tier0, tier1 ] = productDetails.price_tier_list;
+			const perExtraPagePrice = ( tier1.minimum_price - tier0.minimum_price ) / 100;
```

Comment:
The `100` here is something of a [magic number](https://www.codebyamir.com/blog/software-anti-patterns-magic-numbers). What does it represent? Could we put it in a named variable/constant/function to make that more clear?

The main reason I was wondering is that sometimes I've seen this sort of math used for calculations that don't apply to currencies which do not have a decimal (eg `JPY`) and I just want to make sure it's clear what is happening here.

Reply:
Thanks @sirbrillig I was actually trying to get the answer I was looking for. Thanks for raising this!
Is there a better way to get the actual number to be displayed? Or should I try to resolve this in the backend?
CC - @aneeshd16 @DavidRothstein

Reply:
> Is there a better way to get the actual number to be displayed? 

🤔 that's a good question. I'm not saying your technique is wrong, I just don't understand it. 😅  

Let's see, if I understand correctly, you want to get the price per page that the user has already paid for extra DIFM pages attached to this purchase. We know that the total price they paid was [purchase.amount](https://github.com/Automattic/wp-calypso/blob/c8b35df0d209e20a9b6365468cbc894fad320624/client/lib/purchases/types.ts#L3) but the purchases endpoint doesn't appear to include any information on the breakdown of that price. (Maybe that's something that could be added on the backend?) To determine it, we're looking at the price from the products endpoint for the `wp_difm_lite` product and comparing the price between tiers before dividing by 100. I still don't really understand how that gives the price per page, but it makes me wonder: could the price for the `wp_difm_lite_extra_page` be used for this? (Please forgive me if that's totally wrong. I don't really grok the pricing of DIFM.)

<img width="383" alt="Screen Shot 2022-08-23 at 6 10 55 PM" src="https://user-images.githubusercontent.com/2036909/186275411-730181fe-c857-4fc5-b352-b55e9886af36.png">

Reply:
Ah yes we are actually moving to a tiered pricing system. This product will be removed.
Context:  pdh1Xd-Oi-p2#comment-1229, D85709-code
So I have to do "math" for this to figure out the per price page. As you said maybe there is a way to get the  the per page price for the backend. Maybe @DavidRothstein will have some advice for us 🤔

Reply:
If I understand correctly, we are trying to calculate the cost of the standalone service and the cost of extra pages from the `product` object. However, the product object will change if we update our pricing in the future and will always have the product's current price. So, calculating these prices from the values in the product object will show incorrect prices for earlier purchases. 

We should be calculating this only from the `purchase` object, which (I believe) is immutable. At this time, the `purchase` object returned from the API does not contain the fields necessary to calculate this, so we would need to add these fields on the backend. The `purchase` object should be able to access the billing plan on the backend (also immutable), so this should be doable.

Reply:
Ah right I will have a look into this, thanks for the context @aneeshd16 .
In the mean time we can also consider the approach that you have taken which is somewhat similar but using the `formatCurrency` function

https://github.com/Automattic/wp-calypso/pull/66872/files#diff-d6d276e55159d4aca386ac0391e990943d93b5dc197e6dc3fad4b0840db3f277R660-R665

Reply:
Sure, but we will need to make these calculations with values from the purchases API. Using values from the `product` object will become a blocker for future pricing updates.

Reply:
What I meant was just using this function :D @aneeshd16 

```javascript
formatCurrency( price, currency, {
		isSmallestUnit: true,
} );
```

Reply:
>We should be calculating this only from the `purchase` object, which (I believe) is immutable. At this time, the `purchase` object returned from the API does not contain the fields necessary to calculate this, so we would need to add these fields on the backend.

That sounds right to me too.

Basically the `price_tier_list` property that's currently on the `product` object could have an equivalent on the `purchase` object -- which in the server-side code would probably be built inside the subscription's `set_properties_for_purchases_rest_api()` method using `$this->product->plan` (or something similar) to get the billing plan (and then extract the tier data from the billing plan the same way the `product` endpoints do).  That would ensure all your data is coming from the billing plan associated with this particular subscription.

Reply:
Thanks for the @aneeshd16 @DavidRothstein for the suggestion. I have implemented the backend change as suggested.
D86680-code